### ISSUE TITLE
setuidgid: fatal: unable to run app.go: file does not exist

### ISSUE THREAD

**Issue by manoj150283** [dokku/dokku#1916]
I followed http://www.mircozeiss.com/host-your-golang-app-on-digital-ocean/ to install a go application inside dokku.

However it is showing "unable to run app.go: file does not exist"

manoj@manoj-All-Series:/media/manoj/9fe379fb-1207-4ca3-a9de-c191fbff9689/Projects/Proof Of Concept/golang$ git push dokku master
Counting objects: 58, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (43/43), done.
Writing objects: 100% (58/58), 9.11 KiB | 0 bytes/s, done.
Total 58 (delta 8), reused 0 (delta 0)
-----> Cleaning up...
-----> Building golang from herokuish...
-----> Adding BUILD_ENV to build environment...
-----> Go app detected
-----> Checking Godeps/Godeps.json file.
-----> Using go1.5
-----> Running: godep go install -tags heroku .
-----> Discovering process types
       Procfile declares types -> web
-----> Releasing golang (dokku/golang:latest)...
-----> Deploying golang (dokku/golang:latest)...
-----> DOKKU_SCALE file found (/home/dokku/golang/DOKKU_SCALE)
=====> web=1
-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://dokku.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 10 seconds ...
remote: App container failed to start!!
=====> golang container output:
       setuidgid: fatal: unable to run app.go: file does not exist
=====> end golang container output
remote: Error response from daemon: Cannot kill container 20a937308b9c324a9c3d4140624be50534031cb7b8556e7eb4e19f3236d34416: notrunning: Container 20a937308b9c324a9c3d4140624be50534031cb7b8556e7eb4e19f3236d34416 is not running
remote: Error: failed to kill containers: [20a937308b9c324a9c3d4140624be50534031cb7b8556e7eb4e19f3236d34416]
To dokku@dokku.florahospitality.com:golang
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@dokku.florahospitality.com:golang'

**Procfile**

web: app.go

![image](https://cloud.githubusercontent.com/assets/7291281/13040863/81f70918-d3d7-11e5-8d53-72b271c5c04b.png)

Please advice.

Regards,
- Manoj

**Comment by manoj150283** [comment]
Anyone please help me on this, as I have to host oauth2_proxy inside dokku. 

https://github.com/bitly/oauth2_proxy

Regards,
- Manoj

**Comment by michaelshobbs** [comment]
Does it work on heroku?

**Comment by u2mejc** [comment]
@manoj150283 More to Michael's point, why not use the Heroku example that is known to work with Dokku? https://github.com/heroku/go-getting-started/

The article in question is pretty old now and almost any tech article more than a year old should be read with caution. The author also didn't point out that his project's directory name is `web` till like halfway through, which may impact you running godeps.

**Comment by u2mejc** [comment]
Hi @manoj150283, wanted to follow up and see if you still need more help with this? Were you able to deploy the heroku example?

**Comment by josegonzalez** [comment]
Closing as there hasn't been any followup. Please feel free to comment with further information and we will re-open as necessary.

**Comment by johnsto** [comment]
I just experienced an issue very, very similar to this whilst deploying a Go package. This same package had previously been deployed without error, but yesterday kept getting the dreaded "setuidgid" error and was stumped by it.

The clue was found by entering a bash shell within the container (`docker run -i -t dokku/myprog /bin/bash`) and observing that the executable in `/app/bin` was now named `src` instead of the value specified in the Procfile (`myprog` in the case of `web: myprog`). 

So I tried updating the Procfile to `web: src` instead and pushed again. Same error. But when examining the new container, I found the executable in `/app/bin` was back to being named `myprog`.

I edited the Procfile back to `web: myprog` and pushed again. This time the deploy process succeeded.

I'm not sure why the build process was building an executable called `src` in the first place ('src'  is not a term/folder used in my source repo), or how editing the Procfile to `web: src` and then back again knocked the system back into order, but hopefully this info might help someone with a broader understanding of Dokku and the Golang buildpack!
